### PR TITLE
tree: Cleanup remaining support for implicit revisions

### PR DESCRIPTION
**Pull request by alex-pardes**
## Description

Removed remaining code supporting implicit revisions and fixed some issues exposed by this.

### CODE REVIEW

File: packages/test/mocha-test-setup/api-report/mocha-test-setup.api.md [hunk 1]
```diff
@@ -4,8 +4,6 @@
 
 ```ts
```
Comment (yann-achard-MS) [inline packages/test/mocha-test-setup/api-report/mocha-test-setup.api.md:6]: Unintended changes

File: packages/dds/tree/src/feature-libraries/optional-field/optionalField.ts [hunk 2]
```diff
@@ -215,20 +205,17 @@ export const optionalChangeRebaser: FieldChangeRebaser<OptionalChangeset> = {
...
 		for (const [id, childChange2] of childChanges2ByOriginalId.entries()) {
 			composedChildChanges.push([id, composeChild(undefined, childChange2)]);
 		}
 
-		for (const [src, dst] of change2.moves) {
-			const leg2Src = taggedAtomId(src, revision2);
-			const leg2Dst = taggedAtomId(dst, revision2);
-			const leg1Src = tryGetFromNestedMap(dstToSrc, leg2Src.revision, leg2Src.localId);
+		for (const [leg2Src, leg2Dst] of change2.moves) {
+			const leg1Src = tryGetFromNestedMap(dstToSrc, leg2Src.revision, leg2Dst.localId);
```
Comment (yann-achard-MS) [inline packages/dds/tree/src/feature-libraries/optional-field/optionalField.ts:218]:
```suggestion
			const leg1Src = tryGetFromNestedMap(dstToSrc, leg2Src.revision, leg2Src.localId);
```

Comment (yann-achard-MS) [reply packages/dds/tree/src/feature-libraries/optional-field/optionalField.ts:218]: Yeah, I'm not seeing a test for it.

File: packages/runtime/runtime-definitions/src/test/types/validateRuntimeDefinitionsPrevious.generated.ts [hunk 3]
```diff
@@ -642,6 +642,18 @@ declare function use_old_InterfaceDeclaration_IGarbageCollectionDetailsBase(
 use_old_InterfaceDeclaration_IGarbageCollectionDetailsBase(
     get_current_InterfaceDeclaration_IGarbageCollectionDetailsBase());
 
+/*
```
Comment (yann-achard-MS) [inline packages/runtime/runtime-definitions/src/test/types/validateRuntimeDefinitionsPrevious.generated.ts:645]: What are these changes excatly?

File: packages/dds/tree/src/test/feature-libraries/modular-schema/modularChangeFamily.spec.ts [hunk 4]
```diff
@@ -1514,35 +1514,6 @@ describe("ModularChangeFamily", () => {
 				);
 			});
 		});
-
-		describe("handles implicit and explicit build revision representations", () => {
```
[the remark this answers is not in the document]
Comment (alex-pardes) [reply packages/dds/tree/src/test/feature-libraries/modular-schema/modularChangeFamily.spec.ts:1518]: Done

File: packages/dds/tree/src/feature-libraries/optional-field/optionalField.ts (hunk 2 above)
Comment (alex-pardes) [reply packages/dds/tree/src/feature-libraries/optional-field/optionalField.ts]: Fixed and added a test

**Review by yann-achard-MS — APPROVED** [review]

**Review by yann-achard-MS — APPROVED** [review]

**Pull request merged**